recipe/build.sh

```diff
@@ -1,146 +1,76 @@
-            --with-szlib="${PREFIX}" \
-            --with-pthread=yes  \
-            ${HDF5_OPTIONS} \
-            --enable-cxx \
-            --enable-fortran \
-            --with-default-plugindir="${PREFIX}/lib/hdf5/plugin" \
-            --enable-threadsafe \
-            --enable-build-mode=production \
-            --enable-unsupported \
-            --enable-hlgiftools=yes \
-            --enable-using-memchecker \
-            --enable-static=no \
-            --enable-ros3-vfd \
-            || (cat config.log; false)
-
-# allow oversubscribing with openmpi in make check
-export OMPI_MCA_rmaps_base_oversubscribe=yes
-# also allow oversubscribing with mvapich
-export MVP_ENABLE_AFFINITY=0
-
-if [[ "$CONDA_BUILD_CROSS_COMPILATION" == 1 ]]; then
-  # parentheses ( make this a sub-shell, so env and cwd changes don't persist
-  # and we can safely unset env vars temporarily
-  (
-    # Make a native build of the hdetect and H5make_libsettings executables
-    mkdir -p native-build/bin
-    pushd native-build/bin
-
-    # MACOSX_DEPLOYMENT_TARGET is for the target_platform and not for build_platform
-    unset MACOSX_DEPLOYMENT_TARGET
-    # openmpi sets up env for cross-compilation by default
-    # clear it all so it's back to native builds
-    export OPAL_PREFIX=$BUILD_PREFIX
-    unset ${!OMPI_@}
-
-    $CC_FOR_BUILD ${SRC_DIR}/fortran/src/H5match_types.c -I ${SRC_DIR}/src/ -o H5match_types
-    $FC_FOR_BUILD ${SRC_DIR}/fortran/src/H5_buildiface.F90 -I ${SRC_DIR}/fortran/src/ -L $BUILD_PREFIX/lib -o H5_buildiface
-    $FC_FOR_BUILD ${SRC_DIR}/hl/fortran/src/H5HL_buildiface.F90 -I ${SRC_DIR}/hl/fortran/src -I ${SRC_DIR}/fortran/src -L $BUILD_PREFIX/lib -o H5HL_buildiface
-  )
-  export PATH=`pwd`/native-build/bin:$PATH
+# We don't have emulation in osx...
+if [[ "$CONDA_BUILD_CROSS_COMPILATION" == 1 && $target_platform == "osx-arm64" ]]; then
+  export H5_SIZEOF_LONG_DOUBLE=8
+  export H5_LDOUBLE_TO_LONG_SPECIAL=NO
+  export H5_LONG_TO_LDOUBLE_SPECIAL=NO
+  export H5_LDOUBLE_TO_LLONG_ACCURATE=YES
+  export H5_LLONG_TO_LDOUBLE_CORRECT=YES
+  export H5_DISABLE_SOME_LDOUBLE_CONV=NO
+  export H5_SYSTEM_SCOPE_THREADS=YES
+  export H5_PAC_FC_MAX_REAL_PRECISION=15
+  export H5_PAC_C_MAX_REAL_PRECISION=17
+  export H5_PAC_FC_ALL_INTEGER_KINDS="{1,2,4,8,16}"
+  export H5_PAC_FC_ALL_REAL_KINDS="{4,8}"
+  export H5_H5CONFIG_F_NUM_IKIND="INTEGER, PARAMETER :: num_ikinds = 5"
+  export H5_H5CONFIG_F_NUM_RKIND="INTEGER, PARAMETER :: num_rkinds = 2"
+  export H5_H5CONFIG_F_IKIND="INTEGER, DIMENSION(1:num_ikinds) :: ikind = (/1,2,4,8,16/)"
+  export H5_H5CONFIG_F_RKIND="INTEGER, DIMENSION(1:num_rkinds) :: rkind = (/4,8/)"
+  export H5_PAC_FORTRAN_NATIVE_INTEGER_SIZEOF="                    4"
+  export H5_PAC_FORTRAN_NATIVE_INTEGER_KIND="           4"
+  export H5_PAC_FORTRAN_NATIVE_REAL_SIZEOF="                    4"
+  export H5_PAC_FORTRAN_NATIVE_REAL_KIND="           4"
+  export H5_PAC_FORTRAN_NATIVE_DOUBLE_SIZEOF="                    8"
```

Comment:
@ajelenak are you saying that this kind of setting of environment variable does not ovewrite

these definitions?
https://github.com/HDFGroup/hdf5/blob/407caf1259dd03685b411f78321f823abd2666a6/config/cmake/HDF5UseFortran.cmake#L424

Reply:
I don't know, have to ask our CMake expert. I have only set CMake variables on the command line (-D option).

Reply:
That would be very helpful in terms of helping the community improve the state of HDF5 support

Reply:
Setting env. variables is not going to affect those CMake variables. Options are setting them on the command line or in a custom cacheinit.cmake file which needs to be specified on the command line.

I know this is not the right place to ask but... is it possible to build conda packages somewhere else? I can build the library in a conda environment on my macOS arm64 laptop without any problem. If all the build tools are available to be installed in a conda environment, would it be possible to use GitHub actions to create conda packages?